apps/mobile/src/components/account-selector-sheet.tsx

```diff
@@ -0,0 +1,213 @@
+import { RefObject, useCallback, useMemo, useRef } from 'react';
+import { ScrollView } from 'react-native';
+import { Gesture, GestureDetector } from 'react-native-gesture-handler';
+import Animated, {
+  SharedValue,
+  runOnJS,
+  useAnimatedStyle,
+  useSharedValue,
+  withTiming,
+} from 'react-native-reanimated';
+
+import { APP_ROUTES } from '@/constants';
+import { useAccounts, userUpdatesAccountOrder } from '@/state/accounts/accounts.slice';
+import { destructAccountIdentifer, useAppDispatch } from '@/state/utils';
+import { BottomSheetModal, BottomSheetScrollView } from '@gorhom/bottom-sheet';
+import { useRouter } from 'expo-router';
+
+import { Box, IconProps } from '@leather.io/ui/native';
+
+import { getAvatarIcon } from './avatar-icon';
+import { Modal } from './bottom-sheet-modal';
+import { AccountCard } from './wallet-settings/account-card';
+
+const APPROXIMATE_DEFAULT_ITEM_SIZE = { height: 150, width: 300 };
+function DraggableAccount({
```

Comment:
Would it make sense to start pulling some of this code apart and using the `features` pattern? Some of these files are quite large and hard to follow. Would `account-selector-sheet` be it's own feature with local component files like `draggable-account`? In the extension, we typ try to only have one component per file if possible for readability.

Reply:
Also, will the `draggable` code be reused? 🤔 Maybe it needs to be more generic?

Reply:
Good points 👍 I've been a bit trigger happy approving mobile stuff to increase velocity but as more of us are working on it now we should take extra care to increase the quality

Reply:
Lgtm, thanks for doing! 👍